### PR TITLE
Fixes #35612: Add check that system meets tuning profile requirements

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: lib/foreman_maintain/concerns/os_facts.rb [hunk 1]
```diff
@@ -105,6 +105,15 @@ def centos?
       def el_short_name
         "el#{el_major_version}"
       end
+
+      def memory
+        meminfo = execute('cat /proc/meminfo')
```
Comment (evgeni) [inline lib/foreman_maintain/concerns/os_facts.rb:110]: there has to be a better way to read files in Ruby ;)

File: definitions/checks/foreman/check_tuning_requirements.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      def check_cpu_cores(tuning_profile)
+        tuning_cpu_cores = tuning_sizes[tuning_profile][:cpu_cores]
+        system_cpu_cores = cpu_cores
+
+        unless system_cpu_cores.to_i >= tuning_cpu_cores
+          "The number of CPU cores for the system is #{system_cpu_cores} but the currently configured tuning profile requires #{tuning_cpu_cores}." # rubocop:disable Metrics/LineLength
+        end
+      end
+
+      def tuning_sizes
```
Comment (evgeni) [inline definitions/checks/foreman/check_tuning_requirements.rb:60]: I wonder if we somehow could read this from `hooks/pre_commit/13-tuning.rb`, but for sure is not a blocker here

**Comment by evgeni** [comment]
besides the `cat` part, I like!

File: definitions/checks/foreman/check_tuning_requirements.rb (hunk 2 above)
Comment (ehelms) [reply definitions/checks/foreman/check_tuning_requirements.rb:60]: I thought about that and the current structure of the file and such seemed not a good interface in the present design.

Comment (evgeni) [reply definitions/checks/foreman/check_tuning_requirements.rb:60]: Yeah, I didn't see a good way either, but hoped you'd do!

**Review by evgeni — APPROVED** [review]

**Pull request merged**